src/lua/serializer.c

```diff
@@ -98,6 +116,100 @@ luaL_serializer_copy_options(struct luaL_serializer *dst,
 			     const struct luaL_serializer *src)
 {
 	memcpy(dst, src, offsetof(struct luaL_serializer, end_of_options));
+
+	/*
+	 * Do a deep copy of the encoder key order. Not doing so would lead to
+	 * double free.
+	 */
+	size_t encode_key_order_len = 0;
+	while (src->encode_key_order != NULL &&
+	       src->encode_key_order[encode_key_order_len] != NULL) {
+		encode_key_order_len++;
+	}
+
+	if (encode_key_order_len > 0) {
+		dst->encode_key_order = xcalloc(
+			encode_key_order_len + 1, sizeof(char *));
+
+		for (size_t i = 0; i < encode_key_order_len; i++) {
+			size_t len = strlen(src->encode_key_order[i]);
+			dst->encode_key_order[i] = xcalloc(len + 1,
+							   sizeof(char));
+			memcpy(dst->encode_key_order[i],
+			       src->encode_key_order[i],
+			       len);
+		}
+	} else {
+		dst->encode_key_order = NULL;
+	}
+}
+
+void
+luaL_serializer_free_options(struct luaL_serializer *cfg)
+{
+	if (!cfg) {
+		return;
+	}
```

Comment:
Nit: I would assume that the pointer points to a valid serializer object: we usually do so and we do it for serializers in other functions. We can add an assertion.